network_importer/adapters/netbox_api/models.py

```diff
@@ -33,6 +33,35 @@
 
 LOGGER = logging.getLogger("network-importer")
 
+from diffsync.enum import DiffSyncModelFlags
+
+def _is_tag_present(obj):
+    """Find if tag is present for a given object."""
+    
+    tags = list()
+    for tag in config.SETTINGS.main.model_flag_tags:
+        if tag in obj['tags']:
+            LOGGER.debug(
+                "Tag (%s) found identified for object %s. Marked for diffsync flag assigment.",
+                tag,
+                obj,
+            )
+            return True
+    return False
+
+def apply_model_flag(item, obj):
+    """Helper function for DiffSync Flag assignment"""
+
+    model_flag = config.SETTINGS.main.model_flag_mode
+ 
+    if model_flag and _is_tag_present(obj):
+        LOGGER.debug(
+            "DiffSync model flag (%s) applied to object %s",
+                model_flag,
+                obj,
+            )
+        item.model_flags = model_flag
```

Comment:
I think we should add the flag to `model_flags` instead of overwriting it
```
item.model_flags = item.model_flags + model_flag
```

Reply:
Updated accordingly